hexrd/ui/image_canvas.py

```diff
@@ -998,6 +1002,43 @@ def compute_azimuthal_integral_sum(self, scaled=True):
         masked = np.ma.masked_array(pimg, mask=np.isnan(pimg))
         return masked.sum(axis=0) / np.sum(~masked.mask, axis=0)
 
+    def update_azimuthal_plot_overlays(self):
+        while self.azimuthal_overlay_artists:
+            item = self.azimuthal_overlay_artists.pop(0)
+            item['artist'].remove()
+            item['line'].remove()
+
+        # Apply new, visible overlays
+        tth, sum = HexrdConfig().last_unscaled_azimuthal_integral_data
+        for overlay in HexrdConfig().azimuthal_overlays:
+            if overlay['visible']:
+                material = HexrdConfig().materials[overlay['material']]
+                material.compute_powder_overlay(tth, fwhm=overlay['fwhm'], scale=overlay['scale'])
+                result = material.powder_overlay
+                line, = self.azimuthal_integral_axis.plot(tth, result, lw=0)
+                artist = self.azimuthal_integral_axis.fill_between(
+                    tth,
+                    result,
+                    color=overlay['color'],
+                    alpha=overlay['opacity']
+                )
+                artist.set_label(overlay['name'])
+                self.azimuthal_overlay_artists.append({
+                    'name': overlay['name'],
+                    'material': overlay['material'],
+                    'artist': artist,
+                    'line': line,
+                })
+        if HexrdConfig().azimuthal_legend and len(self.azimuthal_overlay_artists):
+            self.azimuthal_integral_axis.legend()
+        elif self.azimuthal_integral_axis:
+            try:
+                self.azimuthal_integral_axis.get_legend().remove()
+            except:
```

Comment:
Let's not use bare exceptions. Can we either catch specific exceptions or check things with `if` statements?